### ISSUE TITLE
<intrinsic to_fixed_tuple> is not usable in pure-python

### ISSUE THREAD

**Issue by Smit-create** [numba/numba#8812]
I was trying to use pure python function by disabling JIT and came across the following bug:

```
File "/Users/thebigbool/repos/aesara/b.py", line 9, in some_func
    numba.np.unsafe.ndarray.to_fixed_tuple(shape, ndim),
  File "/Users/thebigbool/opt/anaconda3/envs/aesara-dev-1/lib/python3.10/site-packages/numba/core/extending.py", line 364, in __call__
    raise NotImplementedError(msg)
NotImplementedError: <intrinsic to_fixed_tuple> is not usable in pure-python
```

It can be reproduced by using the following snippet:

<details>
<summary> Reproduce the bug </summary>
<p>

```py
import numba
import numpy as np
import time

ndim = 3
def some_func(x, shape):
    return np.reshape(
        np.ascontiguousarray(np.asarray(x)),
        numba.np.unsafe.ndarray.to_fixed_tuple(shape, ndim),
    )

arr = np.random.rand(2, 2, 2)
a = time.time()
f = numba.njit(some_func)
print(type(f))
tmp = f(arr, np.array((2, 2, 2)))
b = time.time()
print("Time 1:", b-a)

numba.config.DISABLE_JIT = 1

arr = np.random.rand(2, 2, 2)
a = time.time()
f = numba.njit(some_func)
print(type(f))
tmp = f(arr, np.array((2, 2, 2)))
b = time.time()
print("Time 2:", b - a)

numba.config.DISABLE_JIT = 0
```

</p>
</details>

Is there a way to disable the JIT and use the same function in pure python?

**Comment by apmasell** [comment]
I don't believe so, `to_fixed_tuple` is trying to do type conversion that is specific to Numba's type system, so it doesn't have a meaning in Python. It might be possible to add a fallback function.

**Comment by apmasell** [comment]
After discussion, intrinsics are intentionally set this way since there won't be a general fallback function. If you wish to have one, you can define your fallback function, then use the `@overload` decorator to redirect to a function that uses the intrinsic. https://numba.pydata.org/numba-doc/dev/extending/overloading-guide.html

Given you are performance testing, I'm not sure that is what you're after though.

**Comment by Smit-create** [comment]
> Given you are performance testing, I'm not sure that is what you're after though.

Actually, I'm trying to get the coverage report, and because `numba.njit`  functions aren't actually reported in pytest's coverage, I was trying to figure out some way to tackle this issue. The snippet which I provided in the OP is just for reproducing the error.

**Comment by stuartarchibald** [comment]
Should this be closed as `@intrinsic` is behaving as designed? The feature request for coverage report generation is tracked in #4268.
